### DISCUSSION TITLE
Accidently pushed sub_XXXXXX functions as metadata. How can I delete them?

### DISCUSSION THREAD

**Discussion by ThisIsMyAltAccount** [naim94a/lumen#24] [General]
**Describe the bug**
I accidentally pushed a bunch sub_XXXXXX functions.

**To Reproduce**
1, Select multiple sub_XXXXX in the functions window.
2, Right click
3, Click Lumina
4, Click Push functions metadata

**Expected behavior**
Don't push sub_XXXXXX metadata.

**Server (please complete the following information):**
 - OS: Debian 10
 - Architecture: x86_64

**Client (please complete the following information):**
 - OS: Windows 7
 - Architecture: x86_64
 - IDA Version: 7.3sp3

**Comment by naim94a** [comment]
I don’t think IDA actually sends metadata about ‘sub_’ functions to lumina
servers (afaik a few prefixes are excluded).
You can verify this by querying your database:

select * from funcs where name like ‘sub\_%’

**Reply by ThisIsMyAltAccount** [reply]
After experimenting with a brand new db file, I've determined its not giving them a "name" in the db.

$ psql -U user -d postgres -f lumen/schema.sql

1, Start Lumen

$ pg_dump postgres > postgress_1.sql

```
COPY public.funcs (id, name, len, db_id, chksum, metadata, rank, push_dt, update_dt) FROM stdin;
\.
```


1, Open mirc-7.00.exe
2, Analyze
3, Find function ```sub_5443B0```
4, Right click on function name
5, Click Lumina
6, Click Push functions metadata

Output window:

```
1 new out of 1 (1.000000 ratio)
*** Thank you for your contribution! ***
```


$ pg_dump postgres > postgress_2.sql

```
COPY public.funcs (id, name, len, db_id, chksum, metadata, rank, push_dt, update_dt) FROM stdin;
1		18373	1	\\x...	\\x...	10	2021-04-29 01:14:30.991433-04	2021-04-29 01:14:30.991433-04
\.
```


1, Find function ```sub_445B50```
2, Rename into ```ndocommands```
3, Right click on function name
4, Click Lumina
5, Click Push functions metadata

Output window:
```
1 new out of 1 (1.000000 ratio)
*** Thank you for your contribution! ***

```

$ pg_dump postgres > postgress_3.sql


I opened ```postgress_3.sql``` in a text editor and found this:


```
COPY public.funcs (id, name, len, db_id, chksum, metadata, rank, push_dt, update_dt) FROM stdin;
1		18373	1	\\x...	\\x...	10	2021-04-29 01:14:30.991433-04	2021-04-29 01:14:30.991433-04
2	ndocommands	29450	1	\\x...	\\x...	10	2021-04-29 01:19:18.029028-04	2021-04-29 01:19:18.029028-04
\.
```
DB files and EXE used for this report (about 3 MB): 

https://www.dropbox.com/s/un5vrjh6hzf3x81/ida_function_names_issue.7z?dl=0

**Reply by ThisIsMyAltAccount** [reply]
Fixed via ```DELETE FROM funcs WHERE name ='';```
